lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -1308,6 +1353,13 @@ void Inliner::run() {
     }
   }
 
+  // Delete debug scopes that ended up being unused. Erase them in reverse order
+  // since scopes at the back may have uses on scopes at the front.
+  for (auto scopeOp : llvm::reverse(debugScopes))
+    if (scopeOp.use_empty())
```

Comment:
Not directly related to this PR but can we make debug.scope `Pure`? If it's allowed to delete unused debug scope it sounds OK to consider them as pure operation?

Reply:
Yes excellent point, `Pure` would be a good trait to add. Will do that as a separate PR.